backend/src/file_upload_service.py

```diff
@@ -0,0 +1,63 @@
+from io import TextIOWrapper
+import time
+from fastapi import HTTPException, UploadFile
+import logging
+import uuid
+
+from pypdf import PdfReader
+from src.session.file_uploads import FileUpload, update_session_file_uploads, get_session_file_upload
+
+logger = logging.getLogger(__name__)
+
+MAX_FILE_SIZE = 10*1024*1024
+
+def handle_file_upload(file:UploadFile) -> str:
+
+    if (file.size or 0) > MAX_FILE_SIZE:
+        raise HTTPException(status_code=400, detail=f"File upload must be less than {MAX_FILE_SIZE} bytes")
+
+
+    all_content = ""
+    if ("application/pdf" == file.content_type):
+
+        start_time = time.time()
+        pdf_file = PdfReader(file.file)
+        all_content = ""
+        for page_num in range(len(pdf_file.pages)):
+            page_text = pdf_file.pages[page_num].extract_text()
+            all_content += page_text
+            all_content += "\n"
+
+        # pdfminer.six processing
+        # out_str = StringIO()
+        # extract_text_to_fp(file.file, out_str) #, codec='cp1251', strip_control=True)
```

Comment:
Do we need this? If so can we put a comment explaining why we've left this in?

Reply:
Removed